Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -0,0 +1,192 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+
+import Storage
+
+private let ThumbnailIdentifier = "Thumbnail"
+private let RowIdentifier = "Row"
+private let SeperatorKind = "seperator"
+private let SeperatorIdentifier = "seperator"
+
+class TopSitesPanel: UIViewController {
+    var collection: UICollectionView!
+    var datasource: TopSitesDataSource!
+    var profile: Profile! {
+        didSet {
+            profile.history.get(nil, complete: { (data) -> Void in
+                self.datasource.data = data
+                self.collection.reloadData()
+            })
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        datasource = TopSitesDataSource(data: Cursor(status: .Failure, msg: "Nothing loaded yet"))
```

Comment:
Don't think this should be localized…just looks like a non-user-facing error message to me.